core/src/main/java/com/sequenceiq/cloudbreak/service/ha/HeartbeatService.java

```diff
@@ -94,55 +95,63 @@
-                    }
-                });
-            } catch (ActionWentFailException af) {
-                LOGGER.error("Failed to update the heartbeat timestamp 5 times for node {}: {}", nodeId, af.getMessage());
-                cancelEveryFlowWithoutDbUpdate();
-            }
+        if (organizationMigrationRunner.isFinished()) {
+            if (cloudbreakNodeConfig.isNodeIdSpecified()) {
+                String nodeId = cloudbreakNodeConfig.getId();
+                try {
+                    retryService.testWith2SecDelayMax5Times(() -> {
+                        try {
+                            CloudbreakNode self = cloudbreakNodeRepository.findById(nodeId).orElse(new CloudbreakNode(nodeId));
+                            self.setLastUpdated(clock.getCurrentTime());
+                            cloudbreakNodeRepository.save(self);
+                            return Boolean.TRUE;
+                        } catch (RuntimeException e) {
+                            LOGGER.error("Failed to update the heartbeat timestamp", e);
+                            metricService.incrementMetricCounter(MetricType.HEARTBEAT_UPDATE_FAILED);
+                            throw new ActionWentFailException(e.getMessage());
+                        }
+                    });
+                } catch (ActionWentFailException af) {
+                    LOGGER.error("Failed to update the heartbeat timestamp 5 times for node {}: {}", nodeId, af.getMessage());
+                    cancelEveryFlowWithoutDbUpdate();
+                }
 
-            cancelInvalidFlows();
+                cancelInvalidFlows();
+            }
         }
     }
 
     @Scheduled(initialDelay = 35000L, fixedDelay = 30000L)
     public void scheduledFlowDistribution() {
-        if (cloudbreakNodeConfig.isNodeIdSpecified()) {
-            List<CloudbreakNode> failedNodes = new ArrayList<>();
-            try {
-                failedNodes.addAll(distributeFlows());
-            } catch (TransactionExecutionException e) {
-                LOGGER.info("Failed to distribute the flow logs across the active nodes, somebody might have already done it. Message: {}", e.getMessage());
-            }
-            try {
-                cleanupNodes(failedNodes);
-            } catch (TransactionExecutionException e) {
-                LOGGER.info("Failed to cleanup the nodes, somebody might have already done it. Message: {}", e.getMessage());
-            }
-
-            String nodeId = cloudbreakNodeConfig.getId();
-            Set<String> allMyFlows = flowLogRepository.findAllByCloudbreakNodeId(nodeId).stream().map(FlowLog::getFlowId).distinct().collect(Collectors.toSet());
-            Set<String> newFlows = allMyFlows.stream().filter(f -> runningFlows.get(f) == null).collect(Collectors.toSet());
-            for (String flow : newFlows) {
+        if (organizationMigrationRunner.isFinished()) {
+            if (cloudbreakNodeConfig.isNodeIdSpecified()) {
```

Comment:
Should be combined into one expression

Reply:
I don't agree because these are checking separate things, and shouldn't be washed into one if